src/compliment/sources/class_members.clj

```diff
@@ -152,31 +152,55 @@
        join
        (format "(%s)")))
 
+(defn- class&members->doc-dispatch [[^Class _cl members]]
+  (class (first members)))
+
+(defmulti ^:private class&members->doc #'class&members->doc-dispatch)
+
+(defmethod class&members->doc java.lang.reflect.Method
+  [[^Class cl members]]
+  (let [f-mem (first members)]
+    (str (.getName cl) "." (.getName f-mem)
+         (join
+          (map (fn [^Method member]
+                 (str "\n  " (doc-method-parameters (.getParameterTypes member))
+                      " -> " (type-to-pretty-string (.getReturnType ^Method member))
+                      " (" (Modifier/toString (.getModifiers member)) ")"))
+               members))
+         "\n")))
+
+(defmethod class&members->doc java.lang.reflect.Constructor
+  [[^Class cl members]]
+  (str (.getName cl) ".new"
+       (join
+        (map (fn [^Constructor member]
+               (str "\n  " (doc-method-parameters (.getParameterTypes member))))
+             members))
+       "\n"))
+
+(defmethod class&members->doc java.lang.reflect.Field
+  [[^Class cl members]]
+  (let [f-mem (first members)]
+    (str (.getName cl) "." (.getName f-mem)
+         (str " = " (try (.get ^Field f-mem nil)
+                         (catch Exception e "?"))
+              " (" (type-to-pretty-string (.getType ^Field f-mem)) ")\n"
+              (Modifier/toString (.getModifiers f-mem)))
+         "\n")))
+
 ^{:lite nil}
 (defn create-members-doc
   "Takes a list of members (presumably with the same name) and turns
   them into a docstring."
   [members]
-  (->> members
-       (group-by (fn [^Member m] (.getDeclaringClass m)))
-       (map (fn [[^Class class, members]]
-              (let [^Member f-mem (first members)]
-                (str (.getName class) "." (.getName f-mem)
-                     (if (instance? Field f-mem)
-                       (str " = " (try (.get ^Field f-mem nil)
-                                       (catch Exception e "?"))
-                            " (" (type-to-pretty-string (.getType ^Field f-mem)) ")\n"
-                            (Modifier/toString (.getModifiers f-mem)))
-                       (join
-                        (map (fn [^Method member]
-                               (when (instance? Method member)
-                                 (str "\n  " (doc-method-parameters (.getParameterTypes member))
-                                      " -> " (type-to-pretty-string (.getReturnType ^Method member))
-                                      " (" (Modifier/toString (.getModifiers member)) ")")))
-                             (distinct members))))
-                     "\n"))))
-       (interpose "\n")
-       join))
+  (let [sort-members (fn [[^Class cl members]]
+                       [cl (sort-by #(.getParameterCount %)
```

Comment:
This changes order of signatures from least arguments to most also for existing (static) members sources. This should help with picking the right annotations.